Imaging Preprocessing Scripts: Loris-MRI configurations moved to the front end from the $profile file: Redmine 8893

MounaSafiHarab opened the pull request:
- Imaging insertion scripts configuration options can now be set from the Configuration module and/or Config table in the database, instead of the $profile ("prod") file.

- Please check the Loris-MRI pull request (aces/Loris-MRI#182) for existing projects on how to migrate existing $profile values to the database.

- These settings can not be overwritten in the config.xml (because they are used by Loris-MRI perl scripts)

Related to:
aces/Loris-MRI#182
and
https://github.com/aces/dicom-archive-tools/pull/45

MounaSafiHarab commented:
@driusan 
default values are set as if it is a new install (so I took the current default values in $profileTemplate);
is it Ok to assume that projects will import the values from their prod file? (since I assume I have no way of reading their "prod" file)

driusan commented:
@MounaSafiHarab  Interesting question. I think you're right that we don't have any way to extract them from the prod file (unless you write a script on the Loris-MRI side), so I don't think there's really anything else you could do than just use the defaults.

Is there anything in them that's going to affect the data validity if people forget to manually change them in the config module?

gluneau commented:
Data truncated for column 'Description' 

```
INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'converter', 'If converter is set to dcm2mnc, the c-version of dcm2mnc will be used. If however you want to use any of the various versions of the perl converter dicom_to_minc of mydcm2mnc you will have to specify what it is called and the uploader will try to invoke it', 1, 0, 'text', ID, 'dcm2mnc binary to use when converting', 1 FROM ConfigSettings WHERE Name="imaging_pipeline"

1 row(s) affected, 1 warning(s): 1265 Data truncated for column 'Description' at row 1 Records: 1  Duplicates: 0  Warnings: 1

```

![image](https://cloud.githubusercontent.com/assets/2903505/24371592/47620142-12f9-11e7-8132-cf0bbb4f4c7e.png)

MounaSafiHarab commented:
@driusan 
I added the script to read the values from the $profile to change those "default" values on the Loris-MRI side. It is added to my Loris-MRI pull request, and it is placed in the `tools/` directory of the Loris-MRI codebase.

@gluneau 
I addressed the truncated value.

driusan commented:
@MounaSafiHarab this has conflicts that need to be resolved.

driusan commented:
also, when this is eventually merged it needs to be squashed to ensure that the unnecessary merge commit doesn't get committed into the history.

MounaSafiHarab commented:
@driusan 
yes the default values in the Loris-MRI installer were used here as well.
I agree, this should NOT be merged before the Loris-MRI one is approved and about to be merged!

ridz1208 commented:
@MounaSafiHarab does this still need testing or is it ready to merge after the related PR has been merged ?

MounaSafiHarab commented:
@ridz1208 

This needs testing mostly on the Loris-MRI side (main changes are there not in this patch) which is why I believe @driusan added the Blocked label.

driusan commented:
Yeah, I added the blocked tag because of the 3 interrelated pull requests.

Maybe they can be discussed/merged at an imaging meeting to make sure they go in in the right order and all works together?

MounaSafiHarab commented:
@cmadjar 
as discussed in the imaging meeting today, assigned to you, along with the Loris-MRI ones (specifically to test the DTI pipeline), while @gluneau is focusing on the /data/ being on a different VM (mounted) as is the case for IBIS

MounaSafiHarab commented:
@cmadjar 
@gluneau 

please test at your earliest convenience (i.e. ASAP:)) so I can incorporate your feedback!

gluneau commented on SQL/Archive/17.1/2017-03-13_AddLorisMRIToConfig.sql:
```diff
@@ -0,0 +1,39 @@
+
+
+-- default imaging_pipeline settings
+INSERT INTO Config (ConfigID, Value) SELECT ID, "/PATH/TO/DATA/location" FROM ConfigSettings cs WHERE cs.Name="Loris-MRI Data Directory";
```
cs.Name should be: "dataDirBasepath"

MounaSafiHarab replied:
@gluneau 

thanks!

gluneau commented:
@MounaSafiHarab @driusan tested with IBIS data, and with insertion of a zip.
